### ISSUE TITLE
Templates: add "overrides" folder, split head/foot sections of html.html.twig into head/foot templates

### ISSUE THREAD

**Issue by alisonjo315** [CU-CommunityApps/cwd_project#13]
- [x] Rename "shared" to "overrides"
  * No these words don't mean the same thing, that's ok -- "shared" is less specific anyway (partial templates can be shared), and I think at this point we prefer a folder called "overrides" and a folder called either "partials" or "includes," ya?
  * The only template in "shared" right now is a field template for field_post_date, so that fits for an "overrides" folder.
- [x] Create "partials" folder
- [x] Split head/foot sections of html.html.twig into head/foot templates (just b/c we do this in child themes kinda every time anyway, right?)
  * i.e. site-header, site-footer, cu-header, cu-search
- [x] Document intended use of "overrides" and "partials" sub-folders.

**⚠️ UPDATE:** Footer template will have to wait, because cwd_base puts footer stuff in page.html.twig, so we can't put it "in" html.html.twig (by way of site-footer.html.twig).
* Follow-up issue: #22

**Comment by alisonjo315** [comment]
(wait til after #11)
